contracts/rewards/src/contract/execute.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::HashMap;
+
+use axelar_wasm_std::nonempty::Uint256;
+use cosmwasm_std::Addr;
+
+use crate::{
+    error::ContractError,
+    msg::RewardsParams,
+    state::{Epoch, Store},
+};
+
+pub struct Contract<S>
+where
+    S: Store,
+{
+    pub store: S,
+}
+
+#[allow(dead_code)]
+impl<S> Contract<S>
+where
+    S: Store,
+{
+    /// Returns the current epoch. The current epoch is computed dynamically based on the current
+    /// block height and the epoch duration. If the epoch duration is updated, we store the epoch
+    /// in which the update occurs as the last checkpoint
+    fn get_current_epoch(&mut self, cur_block_height: u64) -> Result<Epoch, ContractError> {
+        let stored_params = self.store.load_params();
+        let epoch_duration: u64 = stored_params.params.epoch_duration.into();
+        let epoch = stored_params.last_updated;
+        if cur_block_height >= epoch.block_height_started + epoch_duration {
+            let elapsed = cur_block_height - epoch.block_height_started;
+            let epoch_num = ((elapsed) / epoch_duration) + epoch.epoch_num;
+            let block_height_started = cur_block_height - ((elapsed) % epoch_duration);
+
+            let new_epoch = Epoch {
+                epoch_num,
+                block_height_started,
+                rewards: epoch.rewards,
```

Comment:
Wouldn't this simply be `epoch.block_height_started + epoch_count * epoch_duration` if you do `let epoch_count = ((cur_block_height - epoch.block_height_started) / epoch_duration);` at line 32?